### DISCUSSION TITLE
When will a new version be released?

### DISCUSSION THREAD

**Discussion by CristiFati** [mhammond/pywin32#2526] [General]
Sorry for bringing it up here.

Could you **please enable discussions for this repo**, so any similar questions can be asked there?

**Comment by Avasam** [comment]
You might have better chance at an answer by asking on the mailing list

> please **do not open github issues for general support requests**, or for problems or questions using the modules in this package - they will be closed. For such issues, please email the [python-win32 mailing list](http://mail.python.org/mailman/listinfo/python-win32) - note that you must be subscribed to the list before posting.

Although I would also prefer GitHub discussions as they're more accessible and more discoverable.

Now that Python 3.13's first release candidate is out, I think it would be a good time to release pywin32-307 in time for Python 3.13.
I'll sortof highjack this issue to list things I'd personally like to see before next release:

## pywin32

### Installation and version support
- https://github.com/mhammond/pywin32/pull/2349 or some variant. As it opens the door to speeding up deployments (wheels can be taken directly from CI artefacts instead of rebuilding everything locally). And other build/setup modernization
- Removing code paths incompatible with Python 3.13: https://github.com/mhammond/pywin32/pull/2312
- Finishing the Python unicode cleanup: https://github.com/mhammond/pywin32/pull/2343
- Another extremely old deprecation: https://github.com/mhammond/pywin32/pull/2344
- New deprecations:
  - https://github.com/mhammond/pywin32/pull/2280
  - https://github.com/mhammond/pywin32/pull/2356

### Documentation and examples
These are reference material, it'd be good if they could be accurate / in a working state:
- Fixing typos: https://github.com/mhammond/pywin32/pull/2300
- Fixing demos:
  - https://github.com/mhammond/pywin32/pull/2314
  - https://github.com/mhammond/pywin32/pull/2345
  - https://github.com/mhammond/pywin32/pull/2291
  - https://github.com/mhammond/pywin32/pull/2288
  - https://github.com/mhammond/pywin32/pull/2290
  - https://github.com/mhammond/pywin32/pull/2281
  
### Runtime fixes
Not necessary for a new release imo, so consider these "nice to have"
- https://github.com/mhammond/pywin32/pull/2337
- https://github.com/mhammond/pywin32/pull/2302
- https://github.com/mhammond/pywin32/pull/2318
- https://github.com/mhammond/pywin32/pull/2320
- https://github.com/mhammond/pywin32/pull/2319
- https://github.com/mhammond/pywin32/pull/2284
- https://github.com/mhammond/pywin32/pull/2316
- https://github.com/mhammond/pywin32/pull/2118

## adodbapi

CC @vernondcole 

### Installation and version support
- Removing code paths incompatible with Python 3.13: https://github.com/mhammond/pywin32/pull/2311

### Documentation and examples
These are reference material, it'd be good if they could be accurate:

- Decide where the adodbapi changelog should go and its version bump: https://github.com/mhammond/pywin32/pull/2352
- Fix misplaced docstrings: https://github.com/mhammond/pywin32/pull/2250 OR https://github.com/mhammond/pywin32/pull/2307
- https://github.com/mhammond/pywin32/pull/2347

### Runtime fixes
- https://github.com/mhammond/pywin32/pull/2324

**Comment by mhammond** [comment]
> Could you **please enable discussions for this repo**, so any similar questions can be asked there?

I'd be happy to do that if there was evidence (or even an inkling) that it would be a place where others (ie, not me) would answer questions - but this specific question is the exact opposite of that - it's really directed specifically to me. So I don't see how enabling discussions would help in this specific case?

It's worth considering that one of the reasons for the delay is my time, so adding another venue for things which end up directed specifically to me makes that problem worse, not better. If you feel opening discussions would enable people like yourself to respond to things so I don't have to, I'd be fine to experiment with that, but this specific example is the exact opposite of that.

Re releases, I do need to apologize for having let that slip. I need a new process to make the releases directly from github CI builds, meaning a number of things in previous releases will not be shipped in new ones - most notably the chm documentation and mapi bindings. I'm about to be away from home for over 6 weeks, so I'm afraid to say a new release probably isn't going to happen for at least that long.

> Now that Python 3.13's first release candidate is out, I think it would be a good time to release pywin32-307 in time for Python 3.13.

I agree this would be great, but as above, I'm not sure how the timing will work out for that.

> I'll sortof highjack this issue to list things I'd personally like to see before next release:

You mean you'd like to see both "in time for 3.13" *and* that list of items in the next release? To be frank though, I don't think anything in that list should block the next release, except maybe items which block 3.13. I'd rather get back to a regular release cadence. It's worth noting the huge number of cleanup PRs actually works against things here - it both makes the actual issues harder to find, and makes the task of going over them quite daunting. In other words, a doubling of the number of PRs in this repo probably tends to make them get looked at slower rather than faster.

**Comment by Avasam** [comment]
> You mean you'd like to see both "in time for 3.13" and that list of items in the next release?

In a perfect world, sure, but I doubt that's realistic. Take it as a wishlist of things I would personally want first _**if**_ you have the time.
For Python 3.13 support only the following are relevant:
- https://github.com/mhammond/pywin32/pull/2311
- https://github.com/mhammond/pywin32/pull/2312

New deprecations and doc fixes are "sooner rather than later", but don't *have to* be in the next release for Python 3.13 support.

>  I'd rather get back to a regular release cadence.

Would love to see that, and get bugfixes + documentation improvements in following releases if that's the case.

> doubling of the number of PRs in this repo probably tends to make them get looked at slower rather than faster.

Sorry, I didn't want to just forget about something by not posting the PRs. ^^"
There's just so much that I found to be fixed or improved as I'm improving the IntelliSense and infos for checkers for downstream users of pywin32, as well as static checks which revealed tons of existing issues (pywin32 has a lot of historical baggage).
Hence my list above if it helps prioritization, and feel free to prioritize other contributors' PRs over mines.

**Comment by mhammond** [comment]
I just released 307 taking the PRs @Avasam mentioned above. I'll let it bake for a week or so to see if the new release process actually works, before getting back to the other PRs - after that I intend making more timely releases - sorry for the huge delay here!

**Comment by glyph** [comment]
> I just released 307 taking the PRs @Avasam mentioned above. I'll let it bake for a week or so to see if the new release process actually works, before getting back to the other PRs - after that I intend making more timely releases - sorry for the huge delay here!

Thanks for doing this!
